Home Images & Description

PrasantiNalla opened the pull request:
Added array of images for homepage and one line description for the images.
We have created a folder to store images of homepage in images folder and added a text file home.txt under home folder.

PrasantiNalla committed:
Home Images & Description

added array of images for homepage and one line description for the images

francescastocco commented on src/components/Home/Home.txt:
```diff
@@ -0,0 +1,32 @@
+Excluding its hazcams and navcams (but including the Mastcam), 
+Curiosity carries 10 instruments to explore the martian environment. The rover’s robotic arm also sports a drill and a scoop.
+
+2. Preservence seeks signs of possible past microbial life in those habitable environments, particularly in special rocks known to preserve signs of life over time.
```
Just a typo here - I think it should be `perseverance`

PrasantiNalla replied:
fixed the spelling and created an array to store the photos and information.

francescastocco reviewed (changes requested):
The info looks good, ticket also requires an array storing the photos and information as well

PrasantiNalla committed:
Merge remote-tracking branch 'origin/mm-1.1-journeyPageSkeleton' into mm-0.10-homepage-displayphoto

removed home.txt files as directly included the info in array

francescastocco commented on src/components/Home/Home.tsx:
```diff
@@ -1,13 +1,27 @@
+import "./Home.scss";
 
 export const Home: React.FunctionComponent = () => {
+  const homeImagesArray = [
+    { imageUrl: "../../images/homeImages/1-CuriosityRoverWithCameras.jpg", imageInfo: "Excluding its hazcams and navcams (but including the Mastcam), Curiosity carries 10 instruments to explore the martian environment. The rover's robotic arm also sports a drill and a scoop." },
+    { imageUrl: "../../images/homeImages/2-PreservenceRover.jpg", imageInfo: " Perseverance seeks signs of possible past microbial life in those habitable environments, particularly in special rocks known to preserve signs of life over time." },
+    { imageUrl: "../../images/homeImages/3-SpiritRover.jpg", imageInfo: "Spirit uncovered strong evidence that Mars was once much wetter than it is now." },
+    { imageUrl: "../../images/homeImages/4-OppurtunityRover.jpg", imageInfo: "Opportunity and its twin Spirit were tasked with studying sites on Mars where conditions may once have been favorable for life." },
+    { imageUrl: "../../images/homeImages/5-OppurtunityRoverwithCameras.jpg", imageInfo: "The Rovers' instruments are state-of-the-art tools for acquiring information about Martian geology, atmosphere, environmental conditions, and potential biosignatures." },
+    { imageUrl: "../../images/homeImages/6-CuriosityRover.jpg", imageInfo: " Curiosity has demonstrated the ability to land a very large, heavy rover to the surface of Mars and to land more precisely in a 12.4-mile (20-kilometer) landing area" },
+    { imageUrl: "../../images/homeImages/7-Mars Olympus Mons.jpg", imageInfo: "Olympus Mons, Mars' highest mountain, stands 25 km tall - two and a half times the height of Mount Everest." },
+    { imageUrl: "../../images/homeImages/8-mars_approach_from_space.jpeg", imageInfo: "NASA engineers prepared intensively for over 45 days to guide Curiosity Rover through Mars' atmosphere and land successfully. " },
+    { imageUrl: "../../images/homeImages/9-PIA24309_-_rover_at_Jezero_-_from_animation.png", imageInfo: "The Mars Perserverance rover touched down in the 45km wide Jezero Crater. The crater is thought to have once been filled with water - hence its name, which means 'lake' in Czech." },
+    { imageUrl: "../../images/homeImages/10-PlanetMars_1.jpg", imageInfo: "Civilisations throughout history have known Mars for its reddish colour, reminiscent of blood. The name Mars comes from the Ancient Roman God of War." }];
```
This looks good, thank you.
Can we create an interface for this object and use this to set the type of homeImagesArray?
eg. 
```
interface homeImage {
    imageUrl: string;
    imageInfo: string;
}
```
Also can we rename the array to homeImages (naming variables should be self-explanatory enough so we don't have to write array or list in the name)?

PrasantiNalla replied:
we have created an interface and renamed array

francescastocco commented on src/components/Home/Home.tsx:
```diff
@@ -1,13 +1,27 @@
-      <section className ="main-image-description"></section>
-    </main>
+      <section className="main-image">
+        <img src={homeImagesArray[0].imageUrl}></img>
+      </section>
+      <section className="main-image-description">{homeImagesArray[0].imageInfo}</section>
```
I think this is for your next ticket - could you add a screenshot of what it looks like on the site and we can just merge them in together?

PrasantiNalla committed:
Moved Images folder & created a homeImages interface

moved images folder outside of src folder to make it accessible without import and added an interface for homeImages array.

francescastocco commented on src/components/App.tsx:
```diff
@@ -2,7 +2,7 @@ import React from "react";
 import { Header } from "./navigation/Header";
 import { Route, Routes } from "react-router-dom";
-import { Home } from "./Home/Home";
+import { Home } from "./home/Home";
```
Can you rename the file back to `Home` instead of `home`? Because git didn't notice the change we had build issues with the import

francescastocco commented on src/components/Home/Home.tsx:
```diff
@@ -1,13 +1,38 @@
 import React from "react";
-import "./Home.scss" 
+import "./Home.scss";
+import { useState } from "react";
```
Can you add this import to line 1?

francescastocco commented on src/components/Home/Home.tsx:
```diff
@@ -1,13 +1,38 @@
+      <section className="main-image">
+        <img src={homeImages[selectedImage].imageUrl}></img>
+      </section>
+      {/* <section className="main-image-description"><p>{homeImages[selectedImage].imageInfo}</p></section>
+      <button onClick={() => { if (selectedImage < 9) { setSelectedImage(selectedImage + 1) } }}>Next</button>
+      <button onClick={() => { if (selectedImage > 0) { setSelectedImage(selectedImage - 1) } }}>Previous</button> */}
```
Do we need this commented out code or can it be removed?

PrasantiNalla replied:
we started working on next changes and didn't want to remove the code.so we have just commented them out so we don't loose them

PrasantiNalla committed:
Home folder name change

changed home folder name back to Home
